enhancements/machine-api/control-plane-set.md

```diff
@@ -0,0 +1,603 @@
+the OnDelete update strategy.
+
+## Design Details
+
+### Open Questions
+
+Some open questions currently exist, but are called out in the Recreate Update Strategy notes.
+
+1. How do we wish to introduce this feature into the cluster, should it be a Technical Preview in the first instance?
+2. Do we want this to be an optional add-on (delivered via OLM) or do we want this to be a default operator for future
+   clusters?
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+This enhancement does not describe removing a feature of OpenShift, this section is not applicable.
+
+### Upgrade / Downgrade Strategy
+
+When the new operator is introduced into the cluster, it will not operate until a ControlPlaneSet resource has been
+created. This means we do not need to worry about upgrades during the introduction of this resource.
+
+### Version Skew Strategy
+
+The ControlPlaneSet Operator relies on the Machine API. This is a stable API within OpenShift and we are not expecting
+changes that will cause version skew issues over the next handful of releases.
+
+### Operational Aspects of API Extensions
+
+We will introduce a new ClusterOperator for the ControlPlaneSetOperator. Within this we will introduce conditions (TBD)
+which describe the state of the operator and the control plane which it manages.
+
+This new ClusterOperator will be the key resource from which support should discover information if they believe the
+ControlPlaneSet to be degraded.
+
+There should be no effect to existing operators or supportability of other components due to this enhancement, in
+particular, this is strictly adding new functionality on top of an existing API, we do not expect it to impact the
+functionality of other APIs.
+
+#### Failure Modes
+
+- A new Machine fails to scale up
+  - In this scenario, we expect the operator to turn degraded to signal that there is an issue with the Control Plane
+    Machines
+  - We expect that the Machine will contain information identifying the issue with the launch request, which can be
+    rectified by the user before they then delete the Machine, once deleted, the ControlPlaneSet will attempt the
+    scale up again
+  - This process should be familiar from dealing with existing Failed Machines
+  - A MachineHealthCheck targeting ControlPlaneMachines could automatically resolve this issue
+- The ControlPlaneSet webhook is not operational
+  - This will prevent creation and update of ControlPlaneSets
+  - We expect these operations to be very infrequent within a cluster once established, so the impact should be
+    minimal
+  - The failure policy will be to fail closed, as such, when the webhook is down, no update operations will succeed
+  - The Kube API server operator already detects failing webhooks and as such will identify the issue early
```

Comment:
Add: the ControlPlaneSet is not critical for cluster operator. Everything done by the operator described here can be done manually to the machine(set) objects.

Reply:
Good suggestion thanks